### PR TITLE
[bitnami/minio] fix secret creation check

### PR DESCRIPTION
**Pull request by c4tz**
### Description of the change

The "Option to use a secret" introduced in `14.7.0` also brought a small bug with it: The [check if the secret should be created](https://github.com/bitnami/charts/blob/minio/14.8.0/bitnami/minio/templates/secrets.yaml#L6) expects a boolean, but ["include" always returns a string ](https://github.com/helm/helm/issues/11231). Because it is not used directly (only in the [minio.createSecret](https://github.com/bitnami/charts/blob/minio/14.8.0/bitnami/minio/templates/_helpers.tpl#L95) template), even setting `auth.useSecret` to `false` will result in the string "false". This string is then interpreted as / casted to a boolean in the corresponding `if`-clause, where it always is translated to `true` because it is not empty. (If you set `auth.useSecret` to an empty string, the secret will not be created)

### Benefits

`auth.useSecret` actually works as expected:
* If set to `true` the secret will be created (if `existingSecret` is empty)
* If set to `false` the secret won't be created

### Possible drawbacks
I don't see any (yet)

### Applicable issues
-

### Additional information
There would also be the option to change all `if (include "minio.createSecret" .)`s so that they check the string's content instead of only for emptiness. But this would create a larger diff, so I chose not to do it that way.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/minio/templates/_helpers.tpl [hunk 1]
```diff
@@ -93,9 +93,9 @@ Get the root password key.
 Return true if a secret object should be created
 */}}
 {{- define "minio.createSecret" -}}
-{{- if .Values.auth.existingSecret -}}
+{{- if or (.Values.auth.existingSecret) (eq .Values.auth.useSecret false) -}}
 {{- else -}}
-    {{- .Values.auth.useSecret -}}
+    {{- true -}}
 {{- end -}}
 {{- end -}}
```
Comment (migruiz4) [inline bitnami/minio/templates/_helpers.tpl:100]: I know this was like this prior to your PR, but could you please simplify this as:
```
{{- define "minio.createSecret" -}}
{{- if and (not .Values.auth.existingSecret) .Values.auth.useSecret -}}
    {{- true -}}
{{- end -}}
```

This was a really nice catch! Although the value `.Values.auth.useSecret` is of type boolean, what happened here is what I consider of the weirdest behaviors with helm.

What happens with this helpers is the following:
```
{{- define "my_helper" -}}
{{- if .var }}
{{- true -}} <-- when '{{if include "my_helper" .}}' is called, this is interpreted as 'true'
{{- else -}}
{{- false -}} <-- when '{{if include "my_helper" .}}' is called, this is also interpreted as 'true'
{{- end -}}
```

The actual solution to this is to:
```
{{- define "my_helper" -}}
{{- if .var }}
{{- true -}} <-- when '{{if include "my_helper" .}}' is called, this is interpreted as 'true'
{{- end -}} 
{{- end -}}  <-- empty return would be interpreted as 'false'
```

In this case, using empty `if` and return value based on input value only makes logic more complex, leading to this bug you catched.

**Review by migruiz4 — COMMENTED** [review]
Nice catch @c4tz!  Could you please take a look at my comment?

File: bitnami/minio/templates/_helpers.tpl (hunk 1 above)
Comment (c4tz) [reply bitnami/minio/templates/_helpers.tpl]: Ah, thank you for providing a more sleek solution, looks even cleaner!

Yes, that's what I tried to say in my PR description. Sorry if that was written in a confusing way. :)

**Comment by c4tz** [comment]
I accidentally "closed" this PR by force-pushing to my repo (wanted to have a clean history without unnecessary clutter). 

Reopening with this comment.

**Comment by c4tz** [comment]
Why is this being closed again? 🤔 It isn't stale at all.

**Comment by c4tz** [comment]
@migruiz4 the publishing of the new version seems to have failed: https://github.com/bitnami/charts/actions/runs/11661918124 :(

Also, the notification for this failure failed (because the Slack channel might be archived).

**Comment by migruiz4** [comment]
Thanks for the heads up! We are currently facing an issue with our CI/CD pipeline, we will rerun the affected workflows once the issues are solved.

**Comment by c4tz** [comment]
Oh, okay! I thought there might be a problem with my commit.

Hope you can figure out a solution for that issue soon and don't have to spend much time on it :)

**Pull request merged**